### ISSUE TITLE
Storage error

### ISSUE THREAD

**Issue by rafirafirafi** [skyguy126/Xposed-SoundCloudDownloader#7]
Can't download, the app tells me not possible to write to destination directory. Even after giving itaccess to storage
Android 6 tested on 2 devices

**Comment by skyguy126** [comment]
On cm or los, you have to enable the storage permission in app settings (found when app icon is held down and is dragged to app info) as well as privacy guard (found in the settings menu).

**Comment by rafirafirafi** [comment]
I am stock Motorola 6.0.1 rooted
Storage is enabled in app properties... I don't have any privacy guard in my settings.

**Comment by skyguy126** [comment]
Can you paste the xposed log here

**Comment by KunalGupta996** [comment]
It says- Grants permission for writing to external storage and retry.
I am on Lenovo X3a40 6.0.1 rooted cs rom-BETA.. storage is enabled in app permissions. I dont have any external storage.. is there any way of downloading songs to internal storage?
I tried all the options given in the module... it still doesn't work.

**Comment by skyguy126** [comment]
The error is a bit misleading, "external storage" refers to the internal storage at `/sdcard/`. Could you post your Xposed log here after the error happens.

**Comment by rafirafirafi** [comment]
the xposed log contains nothing regarding this error. i think you should remove your test regarding EXTERNALSTORAGE in your code..

**Comment by skyguy126** [comment]
This is the part that is doing the check:

```
if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M && context.checkSelfPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
    // show error
} else {
    Toast.makeText(context, "Downloading...", Toast.LENGTH_SHORT).show();
    XposedMod.download(context, url, name);
}
```
Refer to #3, enabling the permission seems to have fixed it. Removing that piece of code causes the app to crash on Android 6 devices which have the storage permission disabled. The code only exists to show a graceful error message instead of the app crashing all together.

I don't have access to a computer with the build set up so I cannot fix anything right now. Maybe run something along the lines of `Log.d("version", Build.VERSION.SDK_INT)` and `Log.d("state", context.checkSelfPermission(Manifest.permission.WRITE_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED))` and post what those individual statements return.

**Comment by skyguy126** [comment]
@KunalGupta996 @rafirafirafi Possible fix with 163e5c4f84207185de3bb43b4cf280632b94ee0d.

**Comment by rafirafirafi** [comment]
i have android 7 now :(
